fix: [Backport] MTTB-1273 Inconsistent scene name in scene event

EmandM opened the pull request:
<!-- Replace this block with what this PR does and why. Describe what you'd like reviewers to know, how you applied the engineering principles, and any interesting tradeoffs made. Delete bullet points below that don't apply, and update the changelog section as appropriate. -->

<!-- Add short version of the JIRA ticket to the PR title (e.g. "feat: new shiny feature [MTT-123]") -->
[MTTB-1278](https://jira.unity3d.com/browse/MTTB-1273)
fixes: #3418 

## Changelog

- Added: A `ScenePath` field on the `SceneEvent` returned by the `OnSceneEvent` callback.
- Fixed: Various bugs around the `OnSceneEvent` callback.

## Testing and Documentation

- Includes unit tests.

<!--  Uncomment and mark items off with a * if this PR deprecates any API:
### Deprecated API
- [ ] An `[Obsolete]` attribute was added along with a `(RemovedAfter yyyy-mm-dd)` entry.
- [ ] An [api updater] was added.
- [ ] Deprecation of the API is explained in the CHANGELOG.
- [ ] The users can understand why this API was removed and what they should use instead.
-->

## Backport

<!-- If this is a backport:
 - Add the following to the PR title: "\[Backport\] ..." .
 - Link to the original PR.
If this needs a backport - state this here
If a backport is not needed please provide the reason why.
If the "Backports" section is not present it will lead to a CI test failure. 
-->

This is a backport of #3458

EmandM committed:
fix: [Backport] MTTB-1273 Inconsistent scene name in scene event

EmandM committed:
Update CHANGELOG.md

NoelStephensUnity commented on com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs:
```diff
@@ -1265,16 +1255,7 @@ private void OnClientUnloadScene(uint sceneEventId)
-            });
-
-            OnUnload?.Invoke(NetworkManager.LocalClientId, sceneName, sceneUnload);
+            InvokeSceneEvents(NetworkManager.LocalClientId, sceneEventData);
```
Does this need to go above the InvokeSceneEvents?

```
            // The only scenes unloaded are scenes that were additively loaded
            sceneEventData.LoadSceneMode = LoadSceneMode.Additive;
```

NoelStephensUnity commented on com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs:
```diff
@@ -1925,16 +1868,7 @@ private void OnClientBeginSync(uint sceneEventId)
-                });
-
-                OnLoad?.Invoke(NetworkManager.LocalClientId, sceneName, loadSceneMode, sceneLoad);
+                InvokeSceneEvents(NetworkManager.LocalClientId, sceneEventData, sceneLoad);
```
Does this need to add the scene path or is that being automatically done?

EmandM replied:
The scenePath is being calculated from the sceneHash inside of `InvokeSceneEvents`

NoelStephensUnity commented on com.unity.netcode.gameobjects/TestHelpers/Runtime/NetcodeIntegrationTest.cs:
```diff
@@ -1736,7 +1760,9 @@ public NetcodeIntegrationTest(HostOrServer hostOrServer)
         {
             var timeoutHelper = assignedTimeoutHelper ?? s_GlobalTimeoutHelper;
-            Assert.False(timeoutHelper.TimedOut, timeOutErrorMessage);
+            var internalError = m_InternalErrorLog.Length > 0 ? $"{timeOutErrorMessage}\n{m_InternalErrorLog}" : timeOutErrorMessage;
```
Did this need to return early if length > 0?

EmandM replied:
I rewrote the functionality here instead of porting it over. Swapped it to match the original PR

NoelStephensUnity reviewed (commented):
Some areas I noticed differences...

NoelStephensUnity committed:
style

Removing the System.Text from StringBuilder since System.Text is included in a using directive.

NoelStephensUnity committed:
update

removing using directive that is not needed.

EmandM committed:
Merge branch 'develop' of https://github.com/Unity-Technologies/com.unity.netcode.gameobjects into backport/fix/mttb-1273/inconsistent-scene-name-in-scene-event

EmandM committed:
Merge branch 'backport/fix/mttb-1273/inconsistent-scene-name-in-scene-event' of https://github.com/Unity-Technologies/com.unity.netcode.gameobjects into backport/fix/mttb-1273/inconsistent-scene-name-in-scene-event

EmandM committed:
Backport fixes

EmandM committed:
Fix OnClientBeginSync

EmandM committed:
Merge branch 'develop' of https://github.com/Unity-Technologies/com.unity.netcode.gameobjects into backport/fix/mttb-1273/inconsistent-scene-name-in-scene-event

NoelStephensUnity reviewed (approved):
*Ding*
🔔 
Looks good Emma!
